src/transformers/models/whisper/processing_whisper.py

```diff
@@ -84,6 +84,14 @@ def batch_decode(self, *args, **kwargs):
         This method forwards all its arguments to WhisperTokenizer's [`~PreTrainedTokenizer.batch_decode`]. Please
         refer to the docstring of this method for more information.
         """
+
+        if isinstance(args[0], dict) and "segments" in args[0]:
+            segments = args[0].pop("segments")
+
+            kwargs = {"segments": segments, **kwargs}
+
+            args = tuple(args[0]["sequences"].unsqueeze(0))
```

Comment:
It's also surprising to me but basically `args[0]["sequences"]` is of shape `(1, 258)`, we unsqueeze to get a tensor of shape `(1,1, 258)`. 

I don't know why and when does it happen exactly but when we enter `batch_decode`,  `sequences` has been squeezed to a tensor of shape (1, 258) which is what we need (otherwise we get an error). 

If you understand what's happening I would like to know!